### DISCUSSION TITLE
Pip-enviroment not found!

### DISCUSSION THREAD

**Discussion by rkmmmutec** [e-m-b-a/emba#1328] [Q&A]
Hi one and all..!! Hope you all are good and doing well.
Kindly help me to resolve the error Pip-enviroment not found!.

   seq - ok
    sleep - ok
    sort - ok
    strings - ok
    tee - ok
    touch - ok
    tr - ok
    uniq - ok
    unzip - ok
    wc - ok

[*] External utils:
    bc - ok
    tree - ok
    unzip - ok
    7z - ok

[!] Sun Sep 29 11:48:03 PM IST 2024 - Test ended on Sun Sep 29 11:48:03 PM IST 2024 and took about 0 days and 00:00:49 

Pip-enviroment not found!

**Comment by m-1-k-3** [comment]
Have you run the installation procedure?

**Reply by rkmmmutec** [reply]
Yes, i followed all the installation procedure.

**Reply by m-1-k-3** [reply]
Is there the EMBA env directory: ./external/emba_venv

**Reply by rkmmmutec** [reply]
Thank you, M-1-K-3, for your kind attention.

The directory you mentioned was already present. I resolved the error by first activating the virtual environment and then proceeding with the installation. I first cloned the embark repository and then, inside the embark directory, I cloned emba. After that, I ran $ sudo ./installer.sh -d. I encountered another error related to Redis and SQL, which I resolved by pulling the necessary Docker images (docker pull mysql:latest and docker pull redis:latest). After that, I continued with $ sudo ./installer.sh -d. Everything is now working fine.

I then attempted to analyze a firmware binary for the TP-Link router (TL-WR840Nv6_EU_0.9.1_4.17_up_boot[200709-rel33396].bin). (Download Link: [TP-Link TL-WR840N Firmware](https://www.tp-link.com/in/support/download/tl-wr840n/#Firmware)). However, the analysis is taking a significant amount of time to complete and generate a report. I have configured the VM to use 4 cores and 8GB of RAM.

Thank you.
